File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.swift
import Foundation

func solution(_ common:[Int]) -> Int {
    if (common[2]-common[1] == common[1]-common[0]) {// 등차수열일 경우
        return common[common.count - 1] + (common[2] - common[1])
    }
    else { // 등비수열일 경우
        return common[common.count - 1] * (common[2] / common[1])
    }
}

File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.java
class Solution {
    public int solution(int[] common) {
        int answer = 0;
        if (common[2]-common[1] == common[1]-common[0]) // 등차수열일 경우
            return common[common.length - 1] + (common[2] - common[1]);
        else // 등비수열일 경우
            return common[common.length - 1] * (common[2] / common[1]);
    }
}